src/crimson/common/errorator.h

```diff
@@ -596,14 +596,14 @@ struct errorator {
     }
 
     template <class FuncT>
-    auto finally(FuncT &&func) {
+    _future finally(FuncT &&func) {
       return this->then_wrapped(
         [func = std::forward<FuncT>(func)](auto &&result) mutable noexcept {
         if constexpr (seastar::is_future<std::invoke_result_t<FuncT>>::value) {
           return ::seastar::futurize_invoke(std::forward<FuncT>(func)).then_wrapped(
             [result = std::move(result)](auto&& f_res) mutable {
             // TODO: f_res.failed()
-            f_res.discard_result();
+            (void)f_res.discard_result();
```

Comment:
Well, this is C-style but, to be honest, `static_cast<void>(f_res.discard_result())` looks verbose.
Perhaps `[[maybe_unused]]` could help.